### PR TITLE
Calculate business day intervals

### PR DESCRIPTION
**Pull request by DaneWeber**
I'm starting this PR even though this first commit doesn't introduce any functionality. I'll add commits to this branch as I proceed.

### CODE REVIEW

**Comment by DaneWeber** [comment]
At this point it's working, mostly, but I need to configure the `business_time` gem because it's working on UTC time and has limited working hours.

**Comment by DaneWeber** [comment]
@jacohen10 - this is ready for review.

I rebased and had to resolve conflicts. I'm not completely clear how a set of my commits are repeated for this.

File: app/models/chore.rb [hunk 1]
```diff
@@ -1,7 +1,24 @@
 class Chore < ApplicationRecord
+  enum interval_type: { calendar_days: 0, business_days: 1 }
+
   validates :name, presence: true
   validates :interval_days, numericality: { only_integer: true, greater_than: 0 }
 
+  def reset_cycle_date
+    self.last_done = Date.today
+    self.cycle_reset = Time.now
+    case
```
Comment (jacohen10) [inline app/models/chore.rb]: can you split this case statement into it's own function. Also, what's the reasoning for returning false and true?

File: app/models/chore.rb [hunk 2]
```diff
@@ -18,4 +35,14 @@ def due_class(due_date = next_due)
       'due-later'
     end
   end
+
+  def add_business_days(start_date:, work_days:)
+    raise ArgumentError, 'start_date must be a Date' unless start_date.instance_of?(Date)
+    raise ArgumentError, 'work_days must be a positive integer' unless work_days.is_a?(Integer) && work_days > 0
+    start_date += 1 if start_date.wday == 6
```
Comment (jacohen10) [inline app/models/chore.rb:45]: what is wday and divmod?

File: test/models/chore_test.rb [hunk 3]
```diff
@@ -5,90 +5,152 @@ class ChoreTest < ActiveSupport::TestCase
   #   assert true
   # end
```
Comment (jacohen10) [inline test/models/chore_test.rb]: we should go through these tests together

**Review by jacohen10 — CHANGES_REQUESTED** [review]

File: app/models/chore.rb (hunk 2 above)
Comment (DaneWeber) [reply app/models/chore.rb:45]: `wday` is a method for the `Date` class that returns the weekday of a given date:
https://ruby-doc.org/stdlib-2.3.1/libdoc/date/rdoc/Date.html#method-i-wday

`divmod` is a method for the `Numeric` class that returns the quotient and remainder as an array:
https://ruby-doc.org/core-2.3.0/Numeric.html#method-i-divmod

File: app/models/chore.rb (hunk 1 above)
Comment (DaneWeber) [reply app/models/chore.rb]: What's the purpose of moving the `case` statement into its own function? While I will certainly do so, and have ideas for where to divide responsibility, I'm not sure what benefit you're aiming for, and thus not sure if I'll be achieving it.

Regarding returning `true` and `false`, I did that essentially as a debug step. Upon reflection, I should probably pull that out.

File: app/models/chore.rb [hunk 4]
```diff
@@ -1,7 +1,24 @@
 class Chore < ApplicationRecord
+  enum interval_type: { calendar_days: 0, business_days: 1 }
```
Comment (DaneWeber) [inline app/models/chore.rb]: @jacohen10  - `enum` defines the `.calendar_days?` and `.business_days` methods.

Comment (DaneWeber) [reply app/models/chore.rb]: http://api.rubyonrails.org/classes/ActiveRecord/Enum.html

File: test/models/chore_test.rb (hunk 3 above)
Comment (DaneWeber) [reply test/models/chore_test.rb]: Scheduled for tomorrow.

File: app/models/chore.rb (hunk 1 above)
Comment (DaneWeber) [reply app/models/chore.rb]: Updated. Please re-review @jacohen10

**Comment by DaneWeber** [comment]
I'm going to create a new PR for the change to the `add_business_days` function.

**Pull request merged**